### PR TITLE
Backoffice - separated reassign component from assign, and turned the components dumb

### PR DESCRIPTION
**Pull request by Omri-Levy**
### Description
Elaborate on the subject, motivation, and context.

### Related issues
 * Provide a link to each related issue.

### Breaking changes
 * Describe the breaking changes that this pull request introduces.

### How these changes were tested
 * Describe the tests that you ran to verify your changes, including devices, operating systems, browsers and versions.

### Examples and references
 * Links, screenshots, and other resources related to this change.

### Checklist
- [] I have read the [contribution guidelines](CONTRIBUTING.md) of this project
- [] I have read the [style guidelines](STYLE_GUIDE.md) of this project
- [] I have performed a self-review of my own code
- [] I have commented my code, particularly in hard-to-understand areas
- [] I have made corresponding changes to the documentation
- [] My changes generate no new warnings and errors
- [] New and existing tests pass locally with my changes

### CODE REVIEW

File: apps/backoffice-v2/src/pages/Entity/components/Case/Case.Actions.tsx [hunk 1]
```diff
@@ -50,54 +46,57 @@ export const Actions: FunctionComponent<IActionsProps> = ({
   showResolutionButtons = true,
 }) => {
   const {
-    onMutateApproveEntity,
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Entity/components/Case/Case.Actions.tsx:53]: Just re-ordered items and grouped by usage.

File: apps/backoffice-v2/src/pages/Entity/components/Case/hooks/useActions/useActionsLogic.tsx [hunk 2]
```diff
@@ -24,10 +24,23 @@ export const ResubmissionReason = {
   FACE_IS_NOT_MATCHING: 'FACE_IS_NOT_MATCHING',
 } as const;
 
-export const useActions = ({ workflowId, fullName }: IUseActions) => {
-  const onSelectNextEntity = useSelectNextEntity();
+export const useActionsLogic = ({ workflowId, fullName }: IUseActions) => {
+  // <Current entity>
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Entity/components/Case/hooks/useActions/useActionsLogic.tsx:28]: Exploring two new conventions I will really appreciate feedback and will revert if decided against the idea. :pray: ;

1. Opening a group of related logic with `<[Subject]>` and closing with `</[Subject]>`. 
2. A suffix of `Logic` for hooks that just encapsulate state and business logic of components.

File: apps/backoffice-v2/src/pages/Entity/components/Case/hooks/useActions/useActionsLogic.tsx [hunk 3]
```diff
@@ -24,10 +24,23 @@ export const ResubmissionReason = {
   FACE_IS_NOT_MATCHING: 'FACE_IS_NOT_MATCHING',
 } as const;
 
-export const useActions = ({ workflowId, fullName }: IUseActions) => {
-  const onSelectNextEntity = useSelectNextEntity();
+export const useActionsLogic = ({ workflowId, fullName }: IUseActions) => {
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Entity/components/Case/hooks/useActions/useActionsLogic.tsx:27]: Must changes are just re-ordering of lines.

File: apps/backoffice-v2/src/pages/Entity/components/Case/hooks/useActions/useActionsLogic.tsx [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+    onMutateRejectEntity,
+    documentToResubmit,
+    resubmissionReason,
+    onDocumentToResubmitChange,
+    onResubmissionReasonChange,
+    debouncedIsLoadingRejectEntity,
+    debouncedIsLoadingApproveEntity,
+    isLoadingApproveEntity,
+    isLoadingRejectEntity,
+  } = useCaseDecisions({
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Entity/components/Case/hooks/useActions/useActionsLogic.tsx:48]: Can also experiment with moving these hooks out directly into `Case.Actions`.

**Review by alonp99 — APPROVED** [review]

**Pull request closed without merging**